frontend/pages/queries/ManageQueriesPage/components/QueriesTable/QueriesTableConfig.tsx

```diff
@@ -246,26 +255,27 @@ const generateTableHeaders = ({
       ),
     },
   ];
-  if (!isOnlyObserver && !isInherited) {
-    tableHeaders.splice(0, 0, {
+  if (!isOnlyObserver && !omitSelectionColumn) {
+    tableHeaders.unshift({
       id: "selection",
-      Header: (cellProps: IHeaderProps): JSX.Element => {
-        const {
-          getToggleAllRowsSelectedProps,
-          toggleAllRowsSelected,
-        } = cellProps;
-        const { checked, indeterminate } = getToggleAllRowsSelectedProps();
+      // TODO - improve typing of IHeaderProps instead of using any
+      // Header: (headerProps: IHeaderProps): JSX.Element => {
+      Header: (headerProps: any): JSX.Element => {
+        const checkboxProps = getConditionalSelectHeaderCheckboxProps({
+          headerProps,
+          checkIfRowIsSelectable: (row) =>
+            (row.original.team_id ?? undefined) === currentTeamId,
```

Comment:
@jacobshandling says, this will work for both teams and all teams for queries page, so it might work for policies page

Reply:
todo: rachel, look into this